From f72e301f57bb5229a7f145c1d6ca059681abf216 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:07:00 -0400
Subject: [powerpc] Print DAR and DSISR on machine check oopses

Message-id: <1410545655-205645-192-git-send-email-dzickus@redhat.com>
Patchwork-id: 94250
O-Subject: [RHEL7 PATCH 191/626] powerpc: Print DAR and DSISR on machine check oopses
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit c54006491dde7d1b8050c5542716b751be92ed80
Author: Anton Blanchard <anton@samba.org>
Date:   Fri Nov 15 15:41:19 2013 +1100

    powerpc: Print DAR and DSISR on machine check oopses

    Machine check exceptions set DAR and DSISR, so print them in our
    oops output.

    Signed-off-by: Anton Blanchard <anton@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/process.c b/arch/powerpc/kernel/process.c
index 7734daf..1676f65 100644
--- a/arch/powerpc/kernel/process.c
+++ b/arch/powerpc/kernel/process.c
@@ -1008,7 +1008,7 @@ void show_regs(struct pt_regs * regs)
  trap = TRAP(regs);
  if ((regs->trap != 0xc00) && cpu_has_feature(CPU_FTR_CFAR))
   printk("CFAR: "REG"\n", regs->orig_gpr3);
- if (trap == 0x300 || trap == 0x600)
+ if (trap == 0x200 || trap == 0x300 || trap == 0x600)
 #if defined(CONFIG_4xx) || defined(CONFIG_BOOKE)
   printk("DEAR: "REG", ESR: "REG"\n", regs->dar, regs->dsisr);
 #else
-- 
1.7.1